cmd/main.go

```diff
@@ -105,6 +123,66 @@ func main() {
 			aggregator.Aggregate(config.Targets, rw)
 		}
 	})
+	mux.HandleFunc("/alive", func(rw http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		rw.WriteHeader(http.StatusOK)
+	})
+	if *dynamicRegistration {
+		mux.HandleFunc("/register", func(rw http.ResponseWriter, r *http.Request) {
+			defer r.Body.Close()
+			err := r.ParseForm()
+			if err != nil {
+				http.Error(rw, "Bad Request", http.StatusBadRequest)
+				return
+			}
+			name := r.Form.Get("name")
+			address := r.Form.Get("address")
+			if name == "" || address == "" {
+				http.Error(rw, "Bad Request", http.StatusBadRequest)
+				return
+			}
+
+			schema := r.Form.Get("schema")
+			if schema == "" {
+				schema = "http"
+			}
+
+			uri := schema + "://" + address
+			target := name + "=" + uri
+			config.Targets = append(config.Targets, target)
+			if *cacheFilePath != "" {
+				saveTargets(config.Targets, *cacheFilePath)
```

Comment:
would it not be better to fail the request instead of the whole service?

Reply:
Do you mean in case of IO failure when saving? 
If so in most cases this would be configuration issue (like file permission) and because this is repeatable operation (each execution will always lead to this execution branch if cacheFilePath is not null) all request would be failing so there is not much difference than failing service and I thinks failing service is much more visible.